### DISCUSSION TITLE
backing-up-files-to-azure-blob-storage-with-azcopy/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#403] [Announcements]
# backing-up-files-to-azure-blob-storage-with-azcopy/

In this post I describe how I used the azcopy command-line tool to backup some photos to Azure Blob Storage

https://andrewlock.net/backing-up-files-to-azure-blob-storage-with-azcopy/

**Comment by kapsiR** [comment]
If you worry about getting blocked from a major cloud provider, I think these type of scans (PhotoDNA, ...) are done by the majority of them.

I can really recommend [restic](https://github.com/restic/restic) for backups.  
It supports various backends including B2 or Azure Blob storage.

**Comment by gandersson** [comment]
The Azure Pricing Calculator is in my experience by far the simplest way to understand the costs of different Azure services. 

https://azure.microsoft.com/en-gb/pricing/calculator/

**Comment by kkoziarski** [comment]
I backup all my google data using this method https://kamilburczyk.medium.com/backing-up-photos-in-aws-s3-glacier-d379757b0889
It's not automatic but works for me. Every quarter I request google takeout and then send it to AWS. I pay ~$0.50 for 200GB per month. 

But this is only backup and glacier tier should not be ever read. If I would like to download it I would pay much more.

**Comment by VincentVerweij** [comment]
Hi Andrew, one year has passed, are you still using this way to backup your Immich data? Or did you choose something else? And what about the cost of using this Azure approach? Did you manage to automate the stuff in your "what's next"? Thanks!

**Reply by andrewlock** [reply]
Short answer is:
- I'm still doing the same thing
- I haven't automated anything else 😅
- the latest versions of Immich create a backup of the db locally - I need to start syncing that too
- Currently it's costing ~£6pm for ~400GB of storage

Hope that helps!

**Reply by VincentVerweij** [reply]
Awesome, thanks for answering!
